### PR TITLE
Adds unique coin sprites

### PR DESCRIPTION
**Pull request by Buggy123**
Part of the Mime Equality series of PRs, this re-adds old coin sprites. Coins without a unique sprite still use the material color method, and any newly added coins materials will use it by default. Does not add any new coins or any functionality, just new sprites (some of which are unused currently).

old:
image
new:
image

:cl: add: The coin press has received a software patch, and will now use unique designs for most material types instead of the default.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
make codersprites nerd

File: code/modules/mining/ores_coins.dm [hunk 1]
```diff
@@ -317,24 +317,25 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
...
 	icon = 'icons/obj/economy.dmi'
 	name = "coin"
-	icon_state = "coin"
+	icon_state = "coin" //This is changed on initialization, the icon state is used to define which coin type to use.
 	flags_1 = CONDUCT_1
 	force = 1
 	throwforce = 2
 	w_class = WEIGHT_CLASS_TINY
 	custom_materials = list(/datum/material/iron = 400)
-	material_flags = MATERIAL_ADD_PREFIX | MATERIAL_COLOR
```
Comment (Fikou) [inline code/modules/mining/ores_coins.dm]: due to my coin system the only thing you need to do to add a new coin is put it in possible things to input in the coin generator
every new coin will now look the same as the default one and it doesnt make sense that a plastic or runite coin is bronze
also two headed coins are iron coins

**Comment by kriskog** [comment]
Those are our old coin sprites :thinking:

**Comment by improvedname** [comment]
>Adds unique coin sprites
People expecting new sprites of all coins
>just adds new mime ore sprites

**Comment by kriskog** [comment]
No you are missing the point, these were literally removed in favor of the greyscaled one just a few months ago. :man_shrugging:

**Comment by Buggy123** [comment]
Oh. In fairness i wasn't aware of what happened to the old coin sprites. Hmm, give me a bit...

**Comment by Qustinnus** [comment]
webedit man re-adds feature we specifically deprecated

**Comment by Qustinnus** [comment]
fuck wrong button

**Comment by Buggy123** [comment]
I'm working on it, i'm re-adding grayscale for spriteless coins and fixing up a few mistakes that ended up in the version that got pushed.

File: code/modules/mining/ores_coins.dm (hunk 1 above)
Comment (Buggy123) [reply code/modules/mining/ores_coins.dm]: Fixed it, now coin types without a unique sprite will instead use the color system by default. This includes any newly-added coins.

**Comment by Buggy123** [comment]
I re-added the greyscale for coins without unique sprites. Also could someone point me to the PR that removed coin sprites originally, so i can see the reasoning there?

**Comment by FantasticFwoosh** [comment]
> Does not add any new coins or any functionality, just new sprites (some of which are unused currently).

This is a downgrade and i would never personally use these. :-1:

**Comment by Buggy123** [comment]
>This is a downgrade and i would never personally use these. 

What, the unused sprites? I was going to use one of them in a later PR, and I thought I'd leave the other in because it is a decent sprite and there's plenty of unused sprites waiting around for a use anyway. If it's a issue I can take them out.

File: code/modules/mining/ores_coins.dm [hunk 2]
```diff
@@ -317,26 +317,29 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
...
+	material_flags = MATERIAL_ADD_PREFIX
 	var/string_attached
 	var/list/sideslist = list("heads","tails")
 	var/cooldown = 0
 	var/value
 	var/coinflip
+	var/coin_icon = "greyscale" //This is used to define which coin type to use. If this isn't supplied, it uses material color.
```
Comment (ninjanomnom) [inline code/modules/mining/ores_coins.dm]: This needs to be part of the material datum if you want a unique sprite for coins. Material subtypes for coins aren't used for printed coins

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by Buggy123** [comment]
Okay, i'm going to rework this a bit and open another one, because i'm approaching this from the wrong direction.

**Pull request closed without merging**